lib/AST/RequirementMachine/Symbol.cpp

```diff
@@ -295,6 +306,30 @@ Symbol Symbol::forGenericParam(GenericTypeParamType *param,
   return symbol;
 }
 
+Symbol Symbol::forShape(RewriteContext &ctx) {
+  llvm::FoldingSetNodeID id;
+  id.AddInteger(unsigned(Kind::Shape));
+
+  void *insertPos = nullptr;
+  if (auto *symbol = ctx.Symbols.FindNodeOrInsertPos(id, insertPos))
```

Comment:
Since this is a singleton, we should store it in a ctx.TheShapeSymbol variable or something, instead of going through the folding set?

Reply:
Are you suggesting moving the code that creates the symbol storage into the `RewriteContext` constructor to assign it into a `RewriteContext::TheShapeSymbol` field, or something else? Only thing is that the `Storage` type is currently only visible in Symbol.cpp. I'm happy to move code around, just want to understand what you're thinking

Reply:
Yeah, I'm thinking adding a `Symbol::Storage *` to RewriteContext. Then instead of calling ctx.Symbols.FindNodeOrInsertPos() on ctx.Symbols you just check if ctx.TheShapeSymbol is non-null

Reply:
Oh I see, that's easier than constructing a `Storage` in RewriteContext, thanks!